packages/core/test/actions.test.ts

```diff
@@ -647,6 +648,7 @@ describe('entry/exit actions', () => {
       expect(flushTracked()).toEqual([
         'exit: A.A2.A2_child',
         'exit: A.A2',
+        'exit: A',
```

Comment:
So strange; I wonder why these were expected to be omitted (not exited) originally. What was our stance on whether to exit the root node or not?

Reply:
> So strange; I wonder why these were expected to be omitted (not exited) originally.

It wasn't expected. When working on https://github.com/statelyai/xstate/pull/3424 I've focused only on one side of the reentrancy (re-entering) but I've missed that I didn't actually fix the other half of the story - that the reentered state has to exit first. 

> What was our stance on whether to exit the root node or not?

In v4 we can exit the root state, in v5 this should become impossible (or at least that's what we've decided in the past months).